doc/src/manual/multi-threading.md

```diff
@@ -223,6 +223,70 @@ julia> a
 
 Note that [`Threads.@threads`](@ref) does not have an optional reduction parameter like [`@distributed`](@ref).
 
+### Using `@threads` without data races
+
+Taking the example of a naive sum
+
+```julia-repl
+julia> function sum_single(a)
+           s = 0
+           for i in a
+               s += i
+           end
+           s
+       end
+sum_single (generic function with 1 method)
+
+julia> sum_single(1:1_000_000)
+500000500000
+```
+
+Simply adding `@threads` exposes a data race with multiple threads reading and writing `s` at the same time.
+```julia-repl
+julia> function sum_multi_bad(a)
+           s = 0
+           Threads.@threads for i in a
+               s += i
+           end
+           s
+       end
+sum_multi_bad (generic function with 1 method)
+
+julia> sum_multi_bad(1:1_000_000)
+70140554652
+```
+
+Note that the result is not `500000500000` as it should be, and will most likely change each evaluation.
+
+To fix this, buffers that are specific to the task may be used to segment the sum into chunks that are race-free.
+
+```julia-repl
+julia> function sum_multi_good(a)
+           buffers = WeakKeyDict{Task,Int}()
+           Threads.@threads for i in a
+               b = get(buffers, current_task(), 0)
+               b += i
+               buffers[current_task()] = b
+           end
+           s = 0
+           for b in values(buffers)
+               s += b
+           end
+           s
+       end
+sum_multi_good (generic function with 1 method)
+
+julia> sum_multi_good(1:1_000_000)
+500000500000
+```
+
+Note that we do not use buffers based on the `threadid()` i.e. `buffers = zeros(Threads.nthreads())` because tasks can
+actually change thread at yield points, known as task migration (docs??), which would introduce data races for such a
+buffering approach.
+
+Another option, which would not be performant for a naive sum like this, is the use of atomic operations on variables shared
```

Comment:
This would probably would be much faster for this particular example however, so it might be confusing in this context here

Reply:
Updatedd